src/Shell_Command.php

```diff
@@ -57,19 +58,33 @@ class Shell_Command extends EE_Command {
 	public function __invoke( $args, $assoc_args ) {
 
 		EE\Utils\delem_log( 'ee shell start' );
-		$args      = auto_site_name( $args, 'shell', '' );
-		$site_name = EE\Utils\remove_trailing_slash( $args[0] );
+		$global_services = [ 'global-nginx-proxy', 'global-db', 'global-redis' ];
+		$service         = \EE\Utils\get_flag_value( $assoc_args, 'service' );
 
-		$site = Site::find( $site_name );
+		if ( ! in_array( $service, $global_services, true ) ) {
+			$args      = auto_site_name( $args, 'shell', '' );
+			$site_name = EE\Utils\remove_trailing_slash( $args[0] );
 
-		if ( ! $site || ! $site->site_enabled ) {
-			EE::error( "Site $site_name does not exist or is not enabled." );
-		}
+			$site = Site::find( $site_name );
```

Comment:
Use [EE\Site\Utils\get_site_info()](https://github.com/EasyEngine/site-command/blob/f395260b7c742f364c50c240acf154e8ee667be6/src/helper/site-utils.php#L90) function here. It will take care of all the checks as well in the if condition below as well.